### ISSUE TITLE
Segfault when running tensorflow.python.ops.nn_ops.fractional_avg_pool_v2

### ISSUE THREAD

**Issue by dmc1778** [tensorflow/tensorflow#59369]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

No

### Source

binary

### Tensorflow Version

2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
Probably due to negative or large argument.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import nn_ops
try:
  arg_0_tensor = tf.random.uniform([3, 30, 50, 3], dtype=tf.float64)
  arg_0 = tf.identity(arg_0_tensor)
  arg_1_0 = -32
  arg_1_1 = -34
  arg_1_2 = -46
  arg_1_3 = True
  arg_1 = [arg_1_0,arg_1_1,arg_1_2,arg_1_3,]
  arg_2 = True
  arg_3 = False
  seed = 341261001
  out = nn_ops.fractional_avg_pool_v2(arg_0,arg_1,arg_2,arg_3,seed=seed,)
except Exception as e:
  print("Error:"+str(e))
```
```


### Relevant log output

```shell
2023-01-21 14:02:26.071375: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-21 14:02:26.188308: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-01-21 14:02:26.781007: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 14:02:26.781205: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 14:02:26.781213: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-01-21 14:02:27.424061: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-21 14:02:27.454021: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 14:02:27.454038: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2023-01-21 14:02:27.454299: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Segmentation fault

```
```
</details>

**Comment by gaikwadrahul8** [comment]
Hi, @nimashiri 

Apologies for the delay and I was able to replicate the issue without any error and for your reference I have added [gist file](https://colab.research.google.com/gist/gaikwadrahul8/3121fb384bd8a2acf2c5c49c672d35fc/fractional_avg_pool_v2.ipynb) It seems like you're passing negative values to `pooling_ratio` argument and Pooling ratio for each dimension of `value`, currently only supports row and col dimension and should be >= 1.0. The first and last elements of `pooling_ratio` argument 
 must be 1.0 because we don't allow pooling on batch and channels dimensions.
 
 You can refer our [official documentation](https://www.tensorflow.org/api_docs/python/tf/nn/fractional_avg_pool) for High level API for `tf.nn.fractional_avg_pool `and [source code](https://github.com/tensorflow/tensorflow/blob/d5b57ca93e506df258271ea00fc29cf98383a374/tensorflow/python/ops/nn_ops.py#L6233-L6293)
 
 I have also replicated on Ubuntu 22.04 with Tensorflow==2.10 and it's working as expected if I'm passing correct arguments to the function, for your reference I have added screenshot below 
 
 
![image](https://user-images.githubusercontent.com/115997457/215266950-699485de-8848-4d7a-afeb-764a8335fb1a.png)

If issue still persists please let us know? or Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ? Thank you!

**Comment by nimashiri** [comment]
> Hi, @nimashiri
> 
> Apologies for the delay and I was able to replicate the issue without any error and for your reference I have added [gist file](https://colab.research.google.com/gist/gaikwadrahul8/3121fb384bd8a2acf2c5c49c672d35fc/fractional_avg_pool_v2.ipynb) It seems like you're passing negative values to `pooling_ratio` argument and Pooling ratio for each dimension of `value`, currently only supports row and col dimension and should be >= 1.0. The first and last elements of `pooling_ratio` argument must be 1.0 because we don't allow pooling on batch and channels dimensions.
> 
> You can refer our [official documentation](https://www.tensorflow.org/api_docs/python/tf/nn/fractional_avg_pool) for High level API for `tf.nn.fractional_avg_pool `and [source code](https://github.com/tensorflow/tensorflow/blob/d5b57ca93e506df258271ea00fc29cf98383a374/tensorflow/python/ops/nn_ops.py#L6233-L6293)
> 
> I have also replicated on `Ubuntu 22.04` with `Tensorflow==2.10 `and it's working as expected if I'm passing correct arguments to the function, for your reference I have added screenshot below
> 
> ![image](https://user-images.githubusercontent.com/115997457/215266950-699485de-8848-4d7a-afeb-764a8335fb1a.png)
> 
> If issue still persists please let us know? or Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ? Thank you!

Hi, thanks for the reply. If we pass correct values, obviously we wouldn't get any error. I am running a fuzzer on TF APIs (the test cases are machine generated), so I am looking for corner cases. Would you please run my test case?

**Comment by nimashiri** [comment]
@gaikwadrahul8 any update? would you please run with my input and confirm whether you encounter segfault?

**Comment by gaikwadrahul8** [comment]
Hi, @nimashiri 

Apologize for the delay and sure I'll try with your input and I'll update the same here soon. Thank you!

**Comment by gaikwadrahul8** [comment]
Hi, @nimashiri 

I have executed the same code with your input on `Ubuntu 22.04` with `Tensorflow==2.10` and I'm getting this error message `Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -32 [Op:FractionalAvgPool]` which is expected because `Pooling ratio` for each dimension of `value`(input Tensor), currently only supports row and column dimension and should be >= 1.0 but you're passing negative value(-32) 

![image](https://user-images.githubusercontent.com/115997457/215476168-987130df-5890-44d2-9db9-b29c3ef03a28.png)

Please let me know, if I'm missing something here ? Thank you!

**Comment by mihaimaruseac** [comment]
@learning-to-play

**Comment by SuryanarayanaY** [comment]
Hi @dmc1778 ,

With TF 2.10V, I could able to replicate segfault as reported by you.Please refer to below log for same.

```
tf2.10) suryanarayanay@surya-ubuntu-22-04:~$ python 59369.py
2023-02-17 11:56:31.690389: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-02-17 11:56:31.818148: I tensorflow/core/util/util.cc:169] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-02-17 11:56:31.854390: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-02-17 11:56:32.551974: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/home/suryanarayanay/miniconda3/envs/tf2.10/lib/:/home/suryanarayanay/miniconda3/envs/tf/lib/:/home/suryanarayanay/miniconda3/envs/tf2.10/lib/
2023-02-17 11:56:32.552094: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/home/suryanarayanay/miniconda3/envs/tf2.10/lib/:/home/suryanarayanay/miniconda3/envs/tf/lib/:/home/suryanarayanay/miniconda3/envs/tf2.10/lib/
2023-02-17 11:56:32.552112: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2.10.0
2023-02-17 11:56:37.365568: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-02-17 11:56:38.672327: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1616] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 38239 MB memory:  -> device: 0, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:04.0, compute capability: 8.0
2023-02-17 11:56:38.673873: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1616] Created device /job:localhost/replica:0/task:0/device:GPU:1 with 38239 MB memory:  -> device: 1, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:05.0, compute capability: 8.0
Segmentation fault (core dumped)
(tf2.10) suryanarayanay@surya-ubuntu-22-04:~$
```
But With TF2.12v No segfault happened and successfully raised intended Error.Please refer to below log for 3 consecutive runs.The issue seems fixed in Tf 2.11v itself where the intended error(same as below) raised successfully.

```
(tf) suryanarayanay@surya-ubuntu-22-04:~$ python 59369.py
2023-02-17 11:54:37.662055: I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-02-17 11:54:37.716519: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-02-17 11:54:38.554392: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2.12.0-rc0
2023-02-17 11:54:45.078069: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1635] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 38201 MB memory:  -> device: 0, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:04.0, compute capability: 8.0
2023-02-17 11:54:45.079716: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1635] Created device /job:localhost/replica:0/task:0/device:GPU:1 with 38201 MB memory:  -> device: 1, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:05.0, compute capability: 8.0
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -32 [Op:FractionalAvgPool]
(tf) suryanarayanay@surya-ubuntu-22-04:~$ python 59369.py
2023-02-17 11:54:53.878765: I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-02-17 11:54:53.933300: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-02-17 11:54:54.774923: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2.12.0-rc0
2023-02-17 11:55:01.337828: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1635] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 38201 MB memory:  -> device: 0, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:04.0, compute capability: 8.0
2023-02-17 11:55:01.339549: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1635] Created device /job:localhost/replica:0/task:0/device:GPU:1 with 38201 MB memory:  -> device: 1, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:05.0, compute capability: 8.0
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -32 [Op:FractionalAvgPool]
(tf) suryanarayanay@surya-ubuntu-22-04:~$ python 59369.py
2023-02-17 11:55:14.218761: I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-02-17 11:55:14.273119: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-02-17 11:55:15.119429: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2.12.0-rc0
2023-02-17 11:55:21.649993: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1635] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 38201 MB memory:  -> device: 0, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:04.0, compute capability: 8.0
2023-02-17 11:55:21.651740: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1635] Created device /job:localhost/replica:0/task:0/device:GPU:1 with 38201 MB memory:  -> device: 1, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:05.0, compute capability: 8.0
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -32 [Op:FractionalAvgPool]
(tf) suryanarayanay@surya-ubuntu-22-04:~
```

**Comment by mihaimaruseac** [comment]
This is resolved then. Thank you!
